http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/GKEInterpreter.scala

```diff
@@ -264,23 +279,69 @@ class GKEInterpreter[F[_]: Parallel: ContextShift: Timer](
   ): F[Unit] =
     for {
       ctx <- ev.ask
-      _ <- logger.info(
-        s"Beginning app creation for app ${params.appId} in project ${params.googleProject} | trace id: ${ctx.traceId}"
-      )
+
+      // Grab records from the database
       dbAppOpt <- KubernetesServiceDbQueries.getActiveFullAppByName(params.googleProject, params.appName).transaction
       dbApp <- F.fromOption(dbAppOpt, AppNotFoundException(params.googleProject, params.appName, ctx.traceId))
-      gkeClusterId = dbApp.cluster.getGkeClusterId
-      namespaceName = dbApp.app.appResources.namespace.name
+
+      app = dbApp.app
+      namespaceName = app.appResources.namespace.name
+      dbCluster = dbApp.cluster
+      gkeClusterId = dbCluster.getGkeClusterId
+
+      _ <- logger.info(
+        s"Beginning app creation for app ${app.appName.value} in cluster ${gkeClusterId.toString} | trace id: ${ctx.traceId}"
+      )
+
+      // Create namespace and secrets
+      _ <- logger.info(
+        s"Creating namespace ${namespaceName.value} and secrets for app ${app.appName.value} in cluster ${gkeClusterId.toString} | trace id: ${ctx.traceId}"
+      )
 
       _ <- kubeService.createNamespace(gkeClusterId, KubernetesNamespace(namespaceName))
       secrets <- getSecrets(namespaceName)
       _ <- secrets.parTraverse(secret =>
         kubeService.createSecret(gkeClusterId, KubernetesNamespace(namespaceName), secret)
       )
-      _ <- logger.info(s"Finished app creation for ${params.appId} | trace id: ${ctx.traceId}")
-      // TODO create svc accts and workload identity roles
-      // TODO helm create galaxy
-      // TODO poll galaxy for creation
+
+      // Create KSA
+      ksaName = KubernetesServiceAccount(s"${app.appName.value}-${config.galaxyAppConfig.serviceAccountSuffix.value}")
+      // TODO populate annotations in KSA for Workload Identity once wb-libs GKE client is fixed
+      ksa = KubernetesModels.KubernetesServiceAccount(ServiceAccountName(ksaName.value), Map.empty)
+
+      _ <- logger.info(
+        s"Creating Kubernetes service account ${ksaName.value} for app  ${app.appName.value} in cluster ${gkeClusterId.toString} | trace id: ${ctx.traceId}"
+      )
+      _ <- kubeService.createServiceAccount(gkeClusterId, ksa, KubernetesNamespace(namespaceName))
+
+      // update KSA in DB
+      _ = appQuery.updateKubernetesServiceAccount(app.id, ksaName).transaction
```

Comment:
should this be `_ <-`? if true, probably should add unit test for this

Reply:
Ahh good catch, should be `<-`. I'll try to add more specs to `GKEInterpreterSpec`

Reply:
Added assertions to `LeoPubsubMessageSubscriberSpec` that the KSA is persisted to the DB